### PR TITLE
Added script and new extensions

### LINKED ISSUE #920: Recommend new extensions for VSCode
image

Although in the book we are asking to install Prettier separately, the Prettier extension is ignored on purpose in Wheel. Have faced a lot of false positives when combined with ESLint. A beginner might go down the rabbit hole of trying to fix the `red squiggly` lines that the Prettier extension throws, which might be a false positive. Source of truth should be the command line and all prettier formatting will be done either ways via Git hooks.

@udai1931 _a Add these extensions to https://github.com/bigbinary/wheel/blob/main/.vscode/extensions.json

### PR DESCRIPTION
**Pull request by udai1931**
Fixes #920, Part of https://github.com/bigbinary/bigbinary-website/issues/2520

### CODE REVIEW

**Comment by udai1931** [comment]
@yedhink _a please check.

File: .scripts/setup_vscode.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env ruby
+
+require "json"
+
+config = `curl -s 'https://raw.githubusercontent.com/bigbinary/wheel/main/.vscode/extensions.json'`
+extensions = JSON.parse(config)["recommendations"]
+
+extensions.each do |extension|
+  output = `code --install-extension #{extension}`
```
Comment (yedhink) [inline .scripts/setup_vscode.sh:9]: We need to check whether `code` or `code-insiders` is a valid executable command, as is it's available under the `PATH` or not. Then only we should continue with this statement.

If either `code` or `code-insiders` is missing, then print the message `Couldn't find an executable for VSCode. Please ensure that the code or code-insiders command is available in your PATH env`.

**Review by yedhink — CHANGES_REQUESTED** [review]
@udai1931 _a Can you please add a demo video for the same? Want to see what happens if certain extensions are already installed and how new extensions are getting installed. Thanks.

**Comment by udai1931** [comment]
@yedhink _a Demo and commands check: https://app.usebubbles.com/6CbjSF1JWd52jEG8a9pw9f/my-video-recording

File: .scripts/setup_vscode.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env ruby
+
+require "json"
+
+code_file_location = `which codes`
```
Comment (yedhink) [inline .scripts/setup_vscode.sh:5]: Had you actually ran this script? Because here it should be `code`.

File: .scripts/setup_vscode.sh [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+code_insiders_file_location = `which code-insiders`
```
Comment (yedhink) [inline .scripts/setup_vscode.sh:6]: Don't use `which`. Refer: https://stackoverflow.com/a/677212

Use: https://stackoverflow.com/a/5471032

File: .scripts/setup_vscode.sh [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+if code_file_location.empty? && code_insiders_file_location.empty?
+  puts "Couldn't find an executable for VSCode. Please ensure that the code or code-insiders command is available in your PATH env"
+  return
+end
+
+config = `curl -s 'https://raw.githubusercontent.com/bigbinary/wheel/main/.vscode/extensions.json'`
+extensions = JSON.parse(config)["recommendations"]
+
+extensions.each do |extension|
+  output = code_file_location.empty? ? `code-insiders --install-extension #{extension}` : `code --install-extension #{extension}`
```
Comment (yedhink) [inline .scripts/setup_vscode.sh:17]: Create the executable outside of this loop so that you can avoid performing this condition each and every time.

**Review by yedhink — CHANGES_REQUESTED** [review]
@udai1931 _a

**Comment by udai1931** [comment]
@yedhink _a please check.

**Pull request merged**